omegatrade-app/frontend/src/app/components/simulation/simulation.component.ts

```diff
@@ -0,0 +1,206 @@
+              }, error => {
+                if (error && error.error && error.error.message) {
+                  this.snackBarService.openSnackBar(error.error.message, '');
+                }
+                this.loader = false;
+              }
+            );
+        }
+      });
+    }
+  }
+
+  simulate(formDirective: FormGroupDirective) {
+    if(this.runningSimulation >= this.maxAllowedSimulation){
+      this.snackBarService.openSnackBar(`Cannot simulate more than ${this.maxAllowedSimulation} simulations`, '');
+      return false;
+    }
+    this.loader = true;
+    this.restService.postData('simulations/start', this.simulateForm.value)
+      .pipe(take(1))
+      .subscribe(
+        response => {
+          if (response && response.success) {
+            this.updateCompanyStatus(this.simulateForm.get('companyId').value, true);
+            this.simulateForm.reset();
+            formDirective.resetForm();
+            this.getSimulations();
+          }
+          this.loader = false;
+        },
+        error => {
+          if (error && error.error && error.error.message) {
+            this.snackBarService.openSnackBar(error.error.message, '');
+          }
+          this.loader = false;
+        });
+  }
+
+  getSimulations() {
+    this.loader = true;
+    this.restService.getData('simulations/list')
+      .pipe(take(1))
+      .subscribe(
+        response => {
+          if (response && response.success) {
+            this.runningSimulation = response.data.length;
+            console.log(this.runningSimulation);
+            this.dataSource = new MatTableDataSource(response.data);
+            this.initializeSortAndPagination();
+            if (response.data && response.data.length > 0) {
+              for (var i = 0; i < response.data.length; i++) {
+                this.updateCompanyStatus(response.data[i].companyId, true)
+              }
+            }
+          }
+          this.loader = false;
+        },
+        error => {
+          if (error && error.error && error.error.message) {
+            this.snackBarService.openSnackBar(error.error.message, '');
+          }
+          this.loader = false;
+        });
+  }
+
+  updateSimulation(simulation) {
+    this.loader = true;
+    const payLoad = { sId: simulation.sId, status: (simulation.status) ? false : true }
+    this.restService.putData(`simulations/update`, payLoad)
+      .pipe(take(1))
+      .subscribe(
+        response => {
+          if (response && response.success) {
+            this.getSimulations();
+            this.snackBarService.openSnackBar(response.message, '');
+          }
+          this.loader = false;
+        },
+        error => {
+          if (error && error.error && error.error.message) {
+            this.snackBarService.openSnackBar(error.error.message, '');
+          }
+          this.loader = false;
+        });
+
+  }
+
+  updateCompanyStatus(id: string, value: boolean = false) {
```

Comment:
The second parameter needs a more logical name than `value`.

Reply:
The above issue has been addressed.